### ISSUE TITLE
Assertion library

### ISSUE THREAD

**Issue by joshgoebel** [exercism/wren#3]
Need to decide which to use.

Ref: https://github.com/wren-lang/wren/wiki/Modules#testing--assertions

- https://github.com/robloach/wren-assert

```js
Assert.equal(5, 5)
Assert.equal(5, 5, "Expected 5 to equal 5")
Assert[5, 5]
Assert[5, 5, "Expected 5 to equal 5"]
```

- https://github.com/EvanHahn/wren-please

```js
Please.equal(123, 123)
Please.notEqual(456, 789)
Please.succeed { "Hello" + "nation" }
Please.beNull(null)
Please.exist(anythingOtherThanNull)
```

-  [Expect](https://github.com/gsmaverick/wren-test/blob/master/src/matchers.wren) from https://github.com/gsmaverick/wren-test

```js
Expect.call("foo".indexOf("bar")).toEqual(-1)
Expect.call(false).toBeFalse
Expect.call(true).not.toBeFalse
Expect.call(fiberWithError).toBeARuntimeError("Error message!")
Expect.call(1).toBe(Num)
Expect.call(1).toEqual(1)
```

**Comment by joshgoebel** [comment]
@iHiD Does Exercism (overall) have a preference towards Assert vs Expect style tests?  I see Javascript is using expect style, but I know you have a LOT of language (so I'm imagining this isn't consistent across all). Personally I've always been a TestUnit vs RSpec kind of person, so I lean towards Assert or Please...  

I just wondered if there is a high level preference in the Exercism org or if each language track kind of just does it's own thing.  I don't think we're mature enough to have any obvious winners (by popularity or something) at the Wren level... If Exercism has a strong preference though that could push me that direction.  Expect style would of course make porting JS tests slightly easier.

If we went with Expect I'd likely fork it to clean up the syntax. I find all the call's quite ugly.

```js
Expect["foo".indexOf("bar")].toEqual(-1)
Expect[false].toBeFalse
Expect[true].not.toBeFalse
Expect[fiberWithError].toBeARuntimeError("Error message!")
Expect[1].toBe(Num)
Expect[1].toEqual(1)
```

**Comment by ErikSchierboom** [comment]
I don't think there is any Exercism-wide preference. Tracks should probably just use what is most idiomatic for their language.

> If we went with Expect I'd likely fork/wrap it to clean up the syntax. I find all the call's quite ugly.

I'd say this would be an argument against using `Expect`. Personally I'd prefer the `Assert` version, but `Please` is fine with me too.

**Comment by iHiD** [comment]
I'd vote for Assert or Please 🙂

**Comment by joshgoebel** [comment]
I went with an Expect lib to match the JS one just to ease the conversion from JS -> Wren... and with a lot of multi-cursor select it was reasonably painless.  If someone wants to change this in the future we could, but in the end I just did the simplest thing to get started and build momentum and then it turned out not so bad.

Continuous improvement vs postponed perfection.

Closing this for now.
